src/dev/serial/uart8250.cc

```diff
@@ -137,11 +147,14 @@ Uart8250::readRbr(Register8 &reg)
     else
         DPRINTF(Uart, "empty read of RX register\n");
 
-    status &= ~RX_INT;
-    platform->clearConsoleInt();
+    clearIntr(RX_INT);
 
     if (device->dataAvailable() && registers.ier.get().rdi)
         scheduleIntr(&rxIntrEvent);
+    else if (rxIntrEvent.scheduled())
+        // Must deschedule a future interrupt if there is no more data to read,
+        // otherwise this would confuse the kernel (iir.id=rx but lsr.rdr=0)
+        deschedule(rxIntrEvent);
```

Comment:
Please use curly braces around multiline if statements. There have been too many bugs that were caused by trying to optimize for 2 characters :)

```suggestion
    else if (rxIntrEvent.scheduled()) {
        // Must deschedule a future interrupt if there is no more data to read,
        // otherwise this would confuse the kernel (iir.id=rx but lsr.rdr=0)
        deschedule(rxIntrEvent);
    }
```